x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.json

```diff
@@ -0,0 +1,974 @@
+      }
+    },
+    {
+      "set" : {
+        "field": "destination.packets",
+        "value": "{{checkpoint.server_outbound_packets}}",
+        "if": "ctx.checkpoint?.server_outbound_packets != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "server.packets",
+        "value": "{{checkpoint.server_outbound_packets}}",
+        "if": "ctx.checkpoint?.server_outbound_packets != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "client.bytes",
+        "value": "{{checkpoint.client_outbound_bytes}}",
+        "if": "ctx.checkpoint?.client_outbound_bytes != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "source.bytes",
+        "value": "{{checkpoint.client_outbound_bytes}}",
+        "if": "ctx.checkpoint?.client_outbound_bytes != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "destination.bytes",
+        "value": "{{checkpoint.server_outbound_bytes}}",
+        "if": "ctx.checkpoint?.server_outbound_bytes != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "server.bytes",
+        "value": "{{checkpoint.server_outbound_bytes}}",
+        "if": "ctx.checkpoint?.server_outbound_bytes != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "destination.port",
+        "value": "{{checkpoint.service}}",
+        "if": "ctx.checkpoint?.service != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "server.port",
+        "value": "{{checkpoint.service}}",
+        "if": "ctx.checkpoint?.service != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "client.port",
+        "value": "{{checkpoint.s_port}}",
+        "if": "ctx.checkpoint?.s_port != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "source.port",
+        "value": "{{checkpoint.s_port}}",
+        "if": "ctx.checkpoint?.s_port != null"
+      }
+    },
+    {
+      "set": {
+        "field": "event.category",
+        "value": "authentication",
+        "if": "ctx.checkpoint?.operation == 'Log In'"
+      }
+    },
+    {
+      "set" : {
+        "field": "event.kind",
+        "value": "alert",
+        "if": "['Prevent', 'Detect', 'Quarantine'].contains(ctx.checkpoint?.rule_action)"
+      }
+    },
+    {
+      "set" : {
+        "field": "event.kind",
+        "value": "event",
+        "if": "ctx.event?.kind == null"
+      }
+    },
+    {
+      "set" : {
+        "field": "event.outcome",
+        "value": "success",
+        "if": "['Accept', 'Allow'].contains(ctx.checkpoint?.rule_action)"
+      }
+    },
+    {
+      "set" : {
+        "field": "event.outcome",
+        "value": "success",
+        "if": "ctx.checkpoint?.audit_status == 'Success'"
+      }
+    },
+    {
+      "set" : {
+        "field": "event.outcome",
+        "value": "failure",
+        "if": "ctx.checkpoint?.audit_status == 'Failure'"
+      }
+    },
+    {
+      "set" : {
+        "field": "event.outcome",
```

Comment:
this should probably be event.type = denied, and even.type is an array

Reply:
Changed to use event.outcome success, since the action performed by the firewall was successful, but also added event.type as denied and connection